src/dmd/errors.d

```diff
@@ -339,6 +370,11 @@ else
 private void verrorPrint(const ref Loc loc, Color headerColor, const(char)* header,
         const(char)* format, va_list ap, const(char)* p1 = null, const(char)* p2 = null)
 {
+    // Ignore errors generated from template-ised druntime hooks, because they
+    // display internal compiler information, which is unnecessary to the user.
+    if (isDruntimeHookError(format, ap))
```

Comment:
Because the result is set to both the original `DeleteExp` and the lowered `CallExp` in espressionsem.d, not outputting any error because of this filter means the `DeleteExp` does not raise errors either. If this is the case, it means that the new `_d_delstruct` hook is inconsistent with the `DeleteExp` from which it's lowered. This should not happen and would be an error in implementing the hook.

Do you think the compiler should handle this?

Reply:
Instead of doing this trick, why not simply gag the errors before doing expressionSemantic on the hook lowering?